File: algorithms/searching/linear search/linearSearch.js
const fs = require("fs");
const values = fs
  .readFileSync("values.txt", "utf8")
  .split(",")
  .map((x) => Number(x));

function linearSearch(values, target) {
  for (let i = 0; i < values.length; i++) {
    if (values[i] === target) {
      console.log(`Found ${target} at index ${i}!`);
      return i;
    }
  }
  console.log("Value not found!");
  return false;
}

linearSearch(values, 1);
linearSearch(values, 99);
linearSearch(values, 1110);

File: algorithms/searching/linear search/linear_search.py
data = open("values.txt", "r")
values = [int(v) for v in data.read().split(",")]

def linear_search(values, target):
    for idx,val in enumerate(values):
        if val == target:
            print(f"Found {target} at index {idx}!")
            return idx
    print("Value not found!")
    return False

linear_search(values, 1)
linear_search(values, 99)
linear_search(values, 1110)